collectors/scm/gitlab/src/main/java/com/capitalone/dashboard/collecteur/DefaultGitlabGitClient.java

```diff
@@ -0,0 +1,262 @@
+package com.capitalone.dashboard.collecteur;
+
+import com.capitalone.dashboard.model.Commit;
+import com.capitalone.dashboard.model.GitlabGitRepo;
+import com.capitalone.dashboard.util.Supplier;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.joda.time.DateTime;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestOperations;
+
+import javax.net.ssl.*;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by benathmane on 23/06/16.
+ */
+
+@Component
+@SuppressWarnings({"PMD"})
+public class DefaultGitlabGitClient implements  GitlabGitClient {
+    private final GitlabSettings gitlabSettings;
+
+    private static final Log LOG = LogFactory.getLog(DefaultGitlabGitClient.class);
+
+    private final RestOperations restOperations;
+
+    private static final String SEGMENT_API = "/api/v3/projects/";
+    private static final String PUBLIC_GITLAB_HOST_NAME = "gitlab.company.com";
+
+    @Autowired
+    public DefaultGitlabGitClient(GitlabSettings gitlabSettings,
+                                       Supplier<RestOperations> restOperationsSupplier) {
+        this.gitlabSettings = gitlabSettings;
+        this.restOperations = restOperationsSupplier.get();
+    }
+
+    private TrustManager[ ] get_trust_mgr() {
```

Comment:
You probably need to add a root certificate to your java store. Trusting arbitrary certs is not a safe solution.